### ISSUE TITLE
Filter out commits by Jenkins git user

### ISSUE THREAD

**Issue by HughePaul** [jenkinsci/gitlab-hook-plugin#28]
Hi, we have an issue where a git push from a Jenkins job is causes an endless loop, as the push triggers a new build.
I have forked and implemented a filter to ignore build_now pushes for a specific configured git user.

Would this be something you'd be interested in a pull request for?

I'd look at having it disabled by default and implementing it across other build causes.

**Comment by javiplx** [comment]
I saw #27 will you prompty closed, and I did reopened [JENKINS-30861](https://issues.jenkins-ci.org/browse/JENKINS-30861), which basically asked for this feature. Actually, I've seen on your repo that you've already made the username configurable, which is something I wanted to ask for, so the answer is yes, either reopen your previous PR (I cannot) or open a new one.

**Comment by HughePaul** [comment]
Yes I meant to open a PR to my own fork to get travis running the tests cleanly before doing an upstream PR.
I'll tidy it up and default the username to empty and do a pull request.

**Comment by HughePaul** [comment]
I'm having problems getting the travis tests to run on ruby versions, it seems to hang.
It completes on the jruby versions.

**Comment by javiplx** [comment]
Recover the ruby tests, and don't worry too much about failures. jruby runs the "unit" tests, but the ruby ones are acceptance tests, and include starting a Jenkins instance. Since a few commits, it tends to constantly timeout on 1.596, but for other Jenkins releases, restarting the job works (sometimes a couple of times).

**Comment by HughePaul** [comment]
I've changed the travis config back to use ruby and jruby.
I added in some development steps to the readme I was using to download and develop from scratch with no existing dev environment. Let me know if that isn't suitable for a merge to the upstream.

**Comment by HughePaul** [comment]
the tests are still failing or timing-out in some versions of ruby.  The failures don't seem related to any code I've changed.

**Comment by HughePaul** [comment]
Hi,
Do you know when a new plugin version will be pushed with these changes?
thanks.
